lib/charms/vault_k8s/v0/vault_kv.py

```diff
@@ -379,6 +379,17 @@ def set_unit_credentials(
                 nonce,
             )
             return
+        standard_secret_id = (
+            secret.id[len("secret:") :] if secret.id.startswith("secret:") else secret.id
+        )
```

Comment:
Please feel free to add comments in the docstring or right here related to why we are doing this thing. I know it's explained in the PR description, but in 6 months if someone reads this code it may not be obvious.